queue.c

```diff
@@ -850,6 +850,12 @@ BaseType_t xQueueGenericSend( QueueHandle_t xQueue,
     }
     #endif
 
+    #if ( configUSE_MUTEXES == 1 && configCHECK_MUTEX_GIVEN_BY_OWNER == 1 )
+        configASSERT( pxQueue->uxQueueType != queueQUEUE_IS_MUTEX ||
+                      pxQueue->u.xSemaphore.xMutexHolder == NULL ||

```

Comment:
I think this check is added [here](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/4e2bf2c639cf769aaf26b742e4c11892189ae057/tasks.c#L4164) already.

The call graph is as below:
```
xQueueGenericSend
   |- prvCopyDataToQueue
       |- #if ( configUSE_MUTEXES == 1 )
           {
                if( pxQueue->uxQueueType == queueQUEUE_IS_MUTEX )
                {
                     |- xTaskPriorityDisinherit
                }
           }
```

Let me know if that is incorrect.